core/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -749,10 +753,10 @@ private boolean allocateUnassigned() {
                     } else if (logger.isTraceEnabled()) {
                         logger.trace("No Node found to assign shard [{}]", shard);
                     }
-                    unassigned.ignoreShard(shard);
+                    changed |= unassigned.ignoreShard(shard, Optional.ofNullable(decision));
```

Comment:
@ywelsch I'm a bit confused by `deciders.canAllocate(node, allocation)` and why it should be used to determine the decision.  `minNode != null` if the decision was either YES or THROTTLE.  If the decision is THROTTLE, then we check to see if allocating to the node is allowed at all for the time being, if not, we put it in the list of throttled nodes so we don't try allocating to it in subsequent iterations.  But that doesn't change the fact that the actual decision for the shard on the current node is THROTTLE.  I'm also not following why the shard would be added to the simulation `ModelNode` if the allocation decision is a NO.